pkg/registries/docker/registry_without_digest_call.go

```diff
@@ -24,8 +25,10 @@ type RegistryWithoutManifestCall struct {
 }
 
 // NewRegistryWithoutManifestCall creates a new basic docker registry without a manifest digest call
-func NewRegistryWithoutManifestCall(integration *storage.ImageIntegration, disableRepoList bool) (*RegistryWithoutManifestCall, error) {
-	dockerRegistry, err := NewDockerRegistry(integration, disableRepoList)
+func NewRegistryWithoutManifestCall(integration *storage.ImageIntegration,
+	disableRepoList bool, metricsHandler *types.MetricsHandler,
+) (*RegistryWithoutManifestCall, error) {
+	dockerRegistry, err := NewDockerRegistry(integration, disableRepoList, metricsHandler)
```

Comment:
I'd suggest to make `metricsHandler` a `CreatorOption`. This way we replace metrics-specific boilerplate with a more generic creator option boilerplate. So next time we add something as Creator option we won't need to change those registries that only propagate options.

Reply:
done